lib/internal/http2/core.js

```diff
@@ -1255,6 +1256,12 @@ function connectionListener(socket) {
     socket.on('timeout', socketOnTimeout);
   }
 
+  // TLS ALPN fallback to HTTP/1.1
+  if ((options.allowHTTP1 === undefined || !!options.allowHTTP1 === true) &&
+    (socket.alpnProtocol === false || socket.alpnProtocol === 'http/1.1')) {
+    return httpConnectionListener.call(this, socket);
+  }
```

Comment:
Do you mean a local variable here, for code clarity? Or adding it to the `options` object somewhere else? I did this hacky comparison to avoid setting an extra variable on every connection. Might be a pointless "optimisation."